docs: added repo build guidelines in README.md (#909)
showdownjs/showdown: README.md

    converter = new showdown.Converter({ extensions: ['myExtension'] });
```

## Building

Building your clone of the repository is easy.
> Prerequesites: [npm](https://www.npmjs.com/package/npm) and [npx](https://www.npmjs.com/package/npx) must be installed.

> 1. run `npm install`: installs all the needed third-party packages into the [node_modules](/node_modules)  folder, based on the contents of [package-lock.json](/package-lock.json) and [package.json](/package.json).

Reviewer: ```suggestion
1. run `npm install`.
```
`npm install` doesn't do any magic here, and if you are talking about developers as the target audience for this section, we can safely omit this kinda  excessive description.

> 1. run `npx grunt build`(see [Gruntfile.js](/Gruntfile.js)). This:

Reviewer: ```suggestion
1. run `npx grunt build` (see [`Gruntfile.js`](/Gruntfile.js)). This command:
```

Author: Sure you want to put backquotes inside a link caption? Sounds a bit strange to me.
And if we do that, I think we want to do the same for the link to the dist folder, just below?

Reviewer: It's just a way to highlight important words, filenames, and commands.
To keep things consistent, it's better to highlight the `dist` folder as well. Thanks for spotting this.

    1. Cleans the repo.
    1. Checks code quality ([JSHint](https://jshint.com/) and [ESLint](https://eslint.org/)).

>     1. Runs tests defined in the repo.

Reviewer: ```suggestion
    1. Runs tests.
```

>     1. Creates the [distributable file](/showdown.js) in the [dist](/dist) folder.
>     1. Creates the [minified file](/showdown.min.js) in the same folder.

Reviewer: ```suggestion
    1. Creates the [distributable](/showdown.js) and [minified](/showdown.min.js) files in the [dist](/dist) folder.
```
